setup-scripts/health-checker.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import discord, requests, traceback, asyncio, os, sys, json, io
+from datetime import datetime, timedelta
+from bot_utils import setup, send_msg, siad, sc_precision
+
+"""
+health-checker reads the /health-check endpoint of the portal and dispatches
+messages to a Discord channel.
+"""
+
+# The default check interval in hours.
+DEFAULT_CHECK_INTERVAL = 1
+
+bot_token = setup()
+client = discord.Client()
+
+
+# get_hostname reads the HOSTNAME from the .env file passed as first argument
+# to the script
+async def get_hostname():
+    if len(sys.argv) > 1:
+        env_file = sys.argv[1]
+    with open(env_file, 'r') as file:
+        for line in file.read().split('\n'):
+            pair = line.split("=")
+            if pair[0] == "HOSTNAME":
+                return pair[1]
+    await send_msg(client, "HOSTNAME not found, cannot check health status", force_notify=True)
+    os.exit(0)
+
+
+# exit_after kills the script if it hasn't exited on its own after `delay` seconds
+async def exit_after(delay):
+    await asyncio.sleep(delay)
+    os._exit(0)
+
+
+@client.event
+async def on_ready():
+    await run_checks()
+    asyncio.create_task(exit_after(3))
+
+
+async def run_checks():
+    print("Running Skynet portal health checks")
+    try:
+        await check_health()
+    except: # catch all exceptions
+        trace = traceback.format_exc()
+        print("[DEBUG] run_checks() failed.")
+        if len(trace) < 1900:
+            await send_msg(client, "```\n{}\n```".format(trace), force_notify=False)
+        else:
+            await send_msg(client, "Failed to run the checks!", file=discord.File(io.BytesIO(trace), filename="failed_checks.log"), force_notify=True)
+
+
+# check_health checks /health-check endpoint and reports recent issues
+async def check_health():
+    print("\nChecking wallet/funds health...")
+
+    try:
+        hostname = await get_hostname()
+        res = requests.get("http://"+hostname+"/health-check")
+    except: # catch all exceptions
+        trace = traceback.format_exc()
+        print("[DEBUG] check_health() failed.")
+        if len(trace) < 1900:
+            await send_msg(client, "```\n{}\n```".format(trace), force_notify=False)
+        else:
+            await send_msg(client, "Failed to run the checks!", file=discord.File(io.BytesIO(trace), filename="failed_checks.log"), force_notify=True)
```

Comment:
```suggestion
            await send_msg(client, "Failed to run portal health checks!", file=discord.File(io.BytesIO(trace), filename="failed_checks.log"), force_notify=True)
```

Reply:
Also, do we want to put those failures in a different file or is it a universal file for failures?

Reply:
It's just the name of the file you upload on Discord. The name of the file doesn't matter, it's just how it's going to be saved on your disk when you download it.